main.c

```diff
@@ -72,7 +72,8 @@ static void init()
    
    /* initialize hardware */
    write_log("Initializing Allegro... ");
-   allegro_init();
+   if (allegro_init() != 0)
```

Comment:
I like this change. If you can drop the other commit, I can commit this. Thanks!